src/Lib/Canvas.vala

```diff
@@ -73,7 +72,49 @@ public class Akira.Lib.Canvas : Goo.Canvas {
 
     public Canvas (Akira.Window window) {
         Object (window: window);
+        // Make The canvas the drop target
+        Gtk.drag_dest_set (this, Gtk.DestDefaults.ALL, targets, Gdk.DragAction.COPY);
+        this.drag_data_received.connect(this.on_drag_data_received);
     }
+    // Making The canvas recive the dropped file
+    private const Gtk.TargetEntry[] targets = {{"text/uri-list",0,0}};
+    private void on_drag_data_received (Gdk.DragContext drag_context, int x, int y, 
+        Gtk.SelectionData data, uint info, uint time)
+    {
+    //loop through list of Files
+    const string[] ACCEPTED_TYPES = {
+        "jpeg",
+        "png",
+        "tiff",
+        "svg",
+        "xml",
+        "gif"
+    };
```

Comment:
This is useless since we have the same exact declaration in the ImageManager.
You should reuse that and avoid duplication.